src/geminim.nim

```diff
@@ -139,8 +139,17 @@ proc processTitanRequest(server: Server, req: Request): Future[Response] {.async
     return response(StatusNotAuthorised, "Token not recognized")
 
   var filePath = req.res.filePath
-  if dirExists(filePath):
-    filePath = filePath / "index.gmi" # assume we want to write index.gmi
+  let (parent, _ )= filePath.splitPath
+
+  if dirExists(parent):
```

Comment:
I think it might be better to do it this way with a check for the directory and a separate call to `createDir`, since `existsOrCreateDir` does not create parent directories, and `createDir` does. Just in case someone wants to upload to, say, `example.com/posts/img/image.png` and even the `posts/` directory doesn't exist yet.

Reply:
In that case, I still think the nested ifs can be avoided.
In fact, I think `createDir` with it's error checking should be enough for making sure the parent exists, since it doesnt fail if the directory already exists, and the cost shouldn't be very different from checking the existence of the parent manually(it still does `dirExists` internally anyway).